src/Rendering/Graph.hs

```diff
@@ -75,23 +79,39 @@ instance (ConstructorName syntax, Foldable syntax) =>
     termAlgebra (In ann syntax) = do
       i <- fresh
       parent <- ask
-      let root = vertex $ TermVertex (fromIntegral i) (T.pack (constructorName syntax)) (converting #? Loc.span ann)
+      let root = vertex $ defMessage
+            & P.vertexId .~ fromIntegral i
+            & P.term .~ T.pack (constructorName syntax)
+            & P.maybe'span .~ (converting #? Loc.span ann)
       subGraph <- foldl' (\acc x -> overlay <$> acc <*> local (const root) x) (pure mempty) syntax
       pure (parent `connect` root `overlay` subGraph)
 
 instance (ConstructorName syntax, Foldable syntax) =>
   ToTreeGraph DiffTreeVertex (DiffF syntax Loc Loc) where
   toTreeGraph d = case d of
-    Merge t@(In (a1, a2) syntax)     -> diffAlgebra t  (Merged   (Just (MergedTerm (T.pack (constructorName syntax)) (ann a1) (ann a2))))
-    Patch (Delete t1@(In a1 syntax)) -> diffAlgebra t1 (Deleted  (Just (DeletedTerm (T.pack (constructorName syntax)) (ann a1))))
-    Patch (Insert t2@(In a2 syntax)) -> diffAlgebra t2 (Inserted (Just (InsertedTerm (T.pack (constructorName syntax)) (ann a2))))
+    Merge t@(In (a1, a2) syntax)     -> diffAlgebra t . DiffTreeVertex'Merged $ defMessage
+                                          & P.term .~ T.pack (constructorName syntax)
+                                          & P.maybe'beforeSpan .~ ann a1
+                                          & P.maybe'afterSpan .~ ann a2
+    Patch (Delete t1@(In a1 syntax)) -> diffAlgebra t1 . DiffTreeVertex'Deleted $ defMessage
+                                          & P.term .~ T.pack (constructorName syntax)
+                                          & P.maybe'span .~ ann a1
+    Patch (Insert t2@(In a2 syntax)) -> diffAlgebra t2 . DiffTreeVertex'Inserted $ defMessage
+                                          & P.term .~ T.pack (constructorName syntax)
+                                          & P.maybe'span .~ ann a2
```

Comment:
I’m additionally wondering if this is a hint that we should be producing some internal, intermediate type that we bridge to the wire types separately. What do you think?

Reply:
I think that's a very good direction to move in general.

I think these graph representations essentially *are* a translation of internal types to output types that are suitable for serialization or display.

Reply:
Yeah, that’s a good point. I think superficially I was meaning “move this lens code somewhere else” but a) that’s a different story altogether, b) it looks like it might be super inconvenient, and c) I really don’t think we ought to hold up this PR just for that.